js/products.js

```diff
@@ -0,0 +1,76 @@
+// find and show selected product
+
+let foundProduct = {};
+let totalWeight = null;
+let sumKcalProduct = null;
+let totalKkal = null;
+
+productSelectElement.addEventListener("input", (event) => {
+  foundProduct = products.find(
+    (product) => product.name === event.target.value
+  );
+  if (foundProduct) {
+    productElement.className = "active";
+    nameProductElement.textContent = foundProduct.name;
+    imageProductElement.src = foundProduct.imageUrl;
+    kcalProductElement.textContent = foundProduct.kcal;
+    proteinsProductElement.textContent = foundProduct.proteins;
+    fatsProductElement.textContent = foundProduct.fats;
+    carbohydratesProductElement.textContent = foundProduct.carbohydrates;
+    updateThis(productSelectElement, null);
+  }
+});
+
+weightProductSelectElement.addEventListener("input", (event) => {
+  const productElementWeight = event.target.value / 100;
+  sumKcalProduct = productElementWeight * foundProduct.kcal;
+  let sumProteinsProduct = productElementWeight * foundProduct.proteins;
+  let sumFatsProduct = productElementWeight * foundProduct.fats;
+  let sumCarbohydratesProduct =
+    productElementWeight * foundProduct.carbohydrates;
+  kcalProductElement.textContent = sumKcalProduct.toFixed(2);
+  proteinsProductElement.textContent = sumProteinsProduct.toFixed(2);
+  fatsProductElement.textContent = sumFatsProduct.toFixed(2);
+  carbohydratesProductElement.textContent = sumCarbohydratesProduct.toFixed(2);
+});
+
+// function findTotaQuantity (totalQuantity, calculatedParameter, wrapperResult) {
+//   totalQuantity += calculatedParameter;
+//   wrapperResult.textContent = totalQuantity;
+//   return totalKkal;
+// }
+
+addProduct.addEventListener("click", function () {
+  totalKkal += sumKcalProduct;
+  totalResult.textContent = totalKkal + " " + "Kkal";
+  return totalKkal;
+});
```

Comment:
Rename addProduct to addProductBtn - it is a button element, I believe this name is more readable.

Reply:
It would be much better to move all the function code into some separate function and name it somehow.
Right now it is not readable.

You have to find the meaning of all the variables in this code: "sumKcalProduct", "totalKkal" and "totalResult".

The best practise is to use not global but local variables and arguments for functions.